proto/spacemesh/v1/smesher_types.proto

```diff
@@ -52,20 +52,23 @@ message CoinbaseResponse {
     AccountId account_id = 1;
 }
 
-// Bit flags for compute engine selection
-enum ComputeEngineFlag {
-    COMPUTE_ENGINE_FLAG_UNSPECIFIED = 0;
-    COMPUTE_ENGINE_FLAG_CPU = 1; // useful for testing on systems without a cuda or vulkan GPU
-    COMPUTE_ENGINE_FLAG_CUDA = 2;
-    COMPUTE_ENGINE_FLAG_VULKAN = 4;
+message GpuInfo {
+    uint32 id = 1; // 0, 1, 2...
+    string model = 2; // e.g. Nvidia GTX 2700
+    ComputeApi compute_api = 3; // A provided compute api
```

Comment:
As discussed, there could be multiple API types per physical device, not just one.

Reply:
Yes, for each API a device supports you will get an entry with a different id, so for example, on a system with a modern nvidia gpu you will get 2 GpuInfos - 1 for nvidia+cuda and 1 for nvidia+vulkun, each with its own id which is good. When user is setting up post he needs to provide an id of 1 of these entries which specifies both the target gpu and the api. So, I think that the proposed api change fully addresses this issue. I don't think we should return a list of apis supported per gpu - just a tuple (id, gpu_info_string, api) ....

Reply:
Maybe GPUInfo is confusing and this should just be called ComputeProvider.

Reply:
I changed GpuInfo to PostComputeProvider - I think this is better now.

Reply:
Got it